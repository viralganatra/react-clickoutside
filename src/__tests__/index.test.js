describe('ClickOutsideHOC', () => {
    it('should render correctly, only accepting a stateless functional component or standard ES6 class component to enhance', () => {

    });

    it('should add a className to the div container if the prop clickOutsideClassName is present', () => {

    });

    it('should pass all props except clickOutsideClassName and onClickOutside to the WrappedComponent', () => {

    });

    it('should only call the onClickOutside function if the mousedown event is outside the component', () => {

    });

    it('should unlisten to the mousedown event when the component has unmounted', () => {

    });
});
