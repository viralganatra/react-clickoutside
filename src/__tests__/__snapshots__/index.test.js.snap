exports[`ClickOutsideHOC should add a className to the div container if the prop clickOutsideClassName is present 1`] = `
<div
  className="test">
  <MockClass />
</div>
`;

exports[`ClickOutsideHOC should pass all props except clickOutsideClassName and onClickOutside to the WrappedComponent 1`] = `
<MockClass
  testProp={1}>
  <div>
    Mock
  </div>
</MockClass>
`;

exports[`ClickOutsideHOC should render correctly, only accepting a stateless functional component or standard ES6 class component to enhance 1`] = `
"Expected \"WrappedComponent\" provided as the first argument to ClickOutsideHOC
        to be a function. Instead, received object."
`;

exports[`ClickOutsideHOC should render correctly, only accepting a stateless functional component or standard ES6 class component to enhance 2`] = `
<div
  style={
    Object {
      "display": "inline-block",
    }
  }>
  <MockClass />
</div>
`;

exports[`ClickOutsideHOC should render correctly, only accepting a stateless functional component or standard ES6 class component to enhance 3`] = `
<div
  style={
    Object {
      "display": "inline-block",
    }
  }>
  <MockStateless />
</div>
`;
